cmd/exec-client/main.go

```diff
@@ -0,0 +1,161 @@
+// exec-client starts the remote executor client.
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	pb "github.com/mlaterman/remote-exec/pkg/proto"
+)
+
+func main() {
+	// Client config flags; gather through env vars/file in the future.
+	cert := flag.String("cert", "certs/client-cert.pem", "The client's public cert.")
+	key := flag.String("key", "certs/client-key.pem", "The client's private key.")
+	cas := flag.String("cas", "certs/server-cert.pem", "The server CA.")
+	address := flag.String("address", "localhost:8334", "The server address")
+
+	// grpc arguements; should be defined through some other library in the future.
+	call := flag.String("call", "status", "The procedure call name. REQUIRED, one of [start,stop,status,output].")
+	id := flag.String("id", "", "The process ID. Required for [stop,status,output].")
+	cmd := flag.String("command", "", "The executable to run. Required for [start].")
+	args := flag.String("args", "", "A comma seperated list of arguments. Optional, used for [start].")
+	cpu := flag.Uint64("cpu", 0, "The CPU time in ms allocated to the process. Optional, used for [start].")
+	mem := flag.Uint64("mem", 0, "The memory quota in bytes allocated to the process. Optional, used for [start].")
+	dio := flag.Uint64("io", 0, "The disk read/write limit in bytes/s allocated to the process. Optional, used for [start].")
+
+	flag.Parse()
+	log.Println("Starting exec-client.")
+	validateFlags(*call)
+
+	log.Printf("config: cert=%s", *cert)
+	log.Printf("config: key=%s", *key)
+	log.Printf("config: cas=%s", *cas)
+	log.Printf("config: address=%s", *address)
+
+	cred, err := creds(*cert, *key, *cas)
+	if err != nil {
+		log.Fatalf("Unable to create tls credentials: %v", err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, *address, grpc.WithTransportCredentials(cred), grpc.WithBlock())
+	if err != nil {
+		log.Fatalf("Unable to connect to server: %v", err)
+	}
+	defer conn.Close()
+
+	client := pb.NewExecutorClient(conn)
+
+	switch *call {
+	case "start":
+		resp, err := client.Start(ctx, startParams(*cmd, *args, *cpu, *mem, *dio))
+		if err != nil {
+			log.Fatalf("Start call failed: %v", err)
+		}
+		log.Printf("Start call succeeded: %v", resp)
```

Comment:
This will just dump the struct to `stdout` right? I imagine you want to output something more human friendly?